test/e2e/specs/editor/blocks/buttons.spec.js

```diff
@@ -123,6 +126,143 @@ test.describe( 'Buttons', () => {
+
+	test( 'can apply custom colors', async ( { editor, page, pageUtils } ) => {
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await page.keyboard.type( 'Content' );
+		await editor.openDocumentSettingsSidebar();
+
+		await page.click(
+			'role=region[name="Editor settings"i] >> role=button[name="Text"i]'
+		);
+		await page.click( 'role=button[name="Custom color picker."i]' );
+		await page.waitForSelector( COLOR_INPUT_FIELD_SELECTOR );
+		await page.click( COLOR_INPUT_FIELD_SELECTOR );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await page.keyboard.type( 'ff0000' );
+
+		await page.click(
+			'role=region[name="Editor settings"i] >> role=button[name="Background"i]'
+		);
+		await page.click( 'role=button[name="Custom color picker."i]' );
+		await page.waitForSelector( COLOR_INPUT_FIELD_SELECTOR );
+		await page.click( COLOR_INPUT_FIELD_SELECTOR );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await page.keyboard.type( '00ff00' );
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:buttons -->
+<div class=\"wp-block-buttons\"><!-- wp:button {\"style\":{\"color\":{\"text\":\"#ff0000\",\"background\":\"#00ff00\"}}} -->
+<div class=\"wp-block-button\"><a class=\"wp-block-button__link has-text-color has-background wp-element-button\" style=\"color:#ff0000;background-color:#00ff00\">Content</a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons -->`
+		);
+	} );
+
+	test( 'can apply named gradient background color', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await page.keyboard.type( 'Content' );
+		await editor.openDocumentSettingsSidebar();
+
+		await page.click(
+			'role=region[name="Editor settings"i] >> role=button[name="Background"i]'
+		);
+		await page.click( 'role=tab[name="Gradient"i]' );
+		await page.click(
+			'role=button[name="Gradient: Vivid cyan blue to vivid purple"i]'
+		);
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:buttons -->
+<div class=\"wp-block-buttons\"><!-- wp:button {\"gradient\":\"vivid-cyan-blue-to-vivid-purple\"} -->
+<div class=\"wp-block-button\"><a class=\"wp-block-button__link has-vivid-cyan-blue-to-vivid-purple-gradient-background has-background wp-element-button\">Content</a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons -->`
+		);
+	} );
+
+	test( 'can apply custom gradient background color', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await page.keyboard.type( 'Content' );
+		await editor.openDocumentSettingsSidebar();
+
+		await page.click(
+			'role=region[name="Editor settings"i] >> role=button[name="Background"i]'
+		);
+		await page.click( 'role=tab[name="Gradient"i]' );
+		await page.click(
+			'role=button[name=/^Gradient control point at position 0% with color code/]'
+		);
```

Comment:
The button to change the point of the gradient actually has the following text:

`Gradient control point at position 0% with color code rgb(6,147,227).`

I used forward matching because rgb values may change with WordPress and Gutenberg updates.

Reply:
> For other themes, the test will fail because the selectors in the gradient palette don’t match.

Is this still required if we use regex matching here?